src/core/RandomVariables.jl

```diff
@@ -118,32 +130,224 @@ mutable struct UntypedVarInfo <: AbstractVarInfo
 end
 VarInfo() = UntypedVarInfo()
 
+###########################
+# Single variable VarInfo #
```

Comment:
Does `SingleVarInfo` only contains a single random variable? If so why does `ranges` required for `SingleVarInfo` and fields like `dists`, `gids` are still vectors?

Reply:
In case we have a vector variable where each element has a different prior distribution, so I think `dists` still needs to be a vector. I think `gids` can be shared however. Not sure about `ranges`, in case we have a matrix variable whose columns are multivariate variables, we probably still need it.

Reply:
Hmmm, I see your point, though I guess we should discourage users to do that. And I'm not sure whether it makes or not for a single multivariate random variable whose different dimension follows different distributions.  But I'm OK with it for now.